test/cluster/basic_test.go

```diff
@@ -38,6 +39,29 @@ func getPorts(num uint32) (node uint32, peer uint32) {
 	return
 }
 
+func agreedLeader(t *testing.T, c *setup.Cluster, activeServers ...int) int {
+	var leaders [3]int
+	var leader int
+
+	for _, srvVal := range activeServers {
+		require.Eventually(t, func() bool {
+			clusterStatusResEnv, err := c.Servers[srvVal].QueryClusterStatus(t)
+			if clusterStatusResEnv != nil {
+				_, leader = c.GetServerByID(clusterStatusResEnv.GetResponse().GetLeader())
+				leaders[srvVal] = leader
+			}
+			return err == nil && clusterStatusResEnv != nil && clusterStatusResEnv.GetResponse().GetLeader() != ""
+		}, 30*time.Second, 100*time.Millisecond)
+	}
+
+	for _, srvVal := range activeServers {
+		if leaders[srvVal] != leader {
+			return -1
+		}
+	}
+	return leader
+}
```

Comment:
make this into an exported method of the cluster: 

```
func (c *setup.Cluster, ) AgreedLeader(t *testing.T, activeServers ...int) int
```

and don't do eventually within, do it from the out side, see next comment
```
require.Eventually(t, func() bool { return c.AgreedLeader(t) >=0 }, 30*time.Second, 100*time.Millisecond)
```